src/components/EntityPage/Participant/Utils/FamilyTable.js

```diff
@@ -3,232 +3,249 @@ import { get, flatMap } from 'lodash';
 import { Link } from 'react-router-dom';
 import sanitize from './sanitize';
 import ParticipantDataTable from './ParticipantDataTable';
-import { HPOLink, SNOMEDLink , MONDOLink,NCITLink} from '../../../Utils/DiagnosisAndPhenotypeLinks';
-
-/*
-Needs to be a class: we're using setState to display the table after the calls to graphql are done to populate the rows
- */
-class FamilyTable extends React.Component {
-  constructor(props) {
-    super(props);
-
-    const participant = props.participant;
-    participant.relationship = "(this participant)";
-
-    const compNode = get(participant, 'family.family_compositions.hits.edges[0].node');
-    this.composition = compNode.composition;
-
-    this.famMembersNodes = [participant].concat(get(compNode, 'family_members.hits.edges', []).map(ele => ele.node))
-
-    this.heads = this.buildHeads(this.famMembersNodes);
-    this.rows = this.buildRows(this.famMembersNodes);
+import {
+  HPOLink,
+  SNOMEDLink,
+  MONDOLink,
+  NCITLink,
+} from '../../../Utils/DiagnosisAndPhenotypeLinks';
+
+const isParticipantSharingDxOrPheno = (dxRow, restOfFamilyIds) => {
+  return restOfFamilyIds.some(memId => Boolean(dxRow[memId]) && dxRow[memId] !== '--');
+};
+
+const getParticipantId = (nodes = []) => {
+  const participant = nodes.find(e => e && e.relationship === '(this participant)');
+  // must always exists
+  return participant.kf_id;
+};
+
+const buildDefaultForMembers = (ids = [], defaultValue = '--') => {
+  return ids.reduce((acc, id) => {
+    return {
+      ...acc,
+      [id]: defaultValue,
+    };
+  }, {});
+};
+
+const baselineWithIds = (leftfield, accessor = '', subheader = false, subaccessor = '') => ids => {
```

Comment:
I think it would make more sense to change the signature to:
```js
const baselineWithIds = ids => (leftfield, accessor = '', subheader = false, subaccessor = '') => {
```
since you always call it from the same function with the same value for `ids`. Just store `baselineWithIds(allIds)` in a variable at the beginning of `buildRows`, you'll remove a lot of function calls from the render path.

Reply:
Good point.